playbooks/ceph.yml

```diff
@@ -366,30 +359,99 @@
-      ansible.builtin.assert:
-        that:
-          - hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network) | length > 0
-        fail_msg: "node {{ item }} doesn't have any interface connected to network {{ cifmw_cephadm_rgw_network }}"
-      loop: "{{ _target_hosts }}"
+    - name: Set RGW Network Properties
       when:
-        - cifmw_cephadm_rgw_network != cifmw_cephadm_storage_network
+        - cifmw_ceph_daemons_layout.rgw_enabled | default(true) | bool
+      block:
+        - name: Set RGW network range to storage network only if it was not provided
+          ansible.builtin.set_fact:
+            cifmw_cephadm_rgw_network: "{{ cifmw_cephadm_storage_network }}"
+          when:
+            - cifmw_cephadm_rgw_network is not defined or
+              cifmw_cephadm_rgw_network | length == 0
+
+        - name: Assert if any EDPM nodes n/w interface is missing in RGW network
+          ansible.builtin.assert:
+            that:
+              - hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network) | length > 0
+            fail_msg: "node {{ item }} doesn't have any interface connected to network {{ cifmw_cephadm_rgw_network }}"
+          loop: "{{ _target_hosts }}"
+          when:
+            - cifmw_cephadm_rgw_network != cifmw_cephadm_storage_network
 
-    - name: Get already assigned RGW IP addresses
-      ansible.builtin.set_fact:
-        ips: "{{ ips | default([]) + [ hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network) | first ] }}"
-      loop: "{{ _target_hosts }}"
+        - name: Get already assigned RGW IP addresses
+          ansible.builtin.set_fact:
+            ips: "{{ ips | default([]) + [ hostvars[item][all_addresses] | ansible.utils.ipaddr(cifmw_cephadm_rgw_network) | first ] }}"
+          loop: "{{ _target_hosts }}"
 
-    # cifmw_cephadm_vip is the VIP reserved in the RGW network
-    - name: Set VIP var as empty string
-      ansible.builtin.set_fact:
-        cifmw_cephadm_vip: ""
+        - name: Set VIP var as empty string
+          ansible.builtin.set_fact:
+            cifmw_cephadm_vip: ""
+          when:
+            - cifmw_cephadm_rgw_vip is undefined
 
-    - name: Process VIP
-      ansible.builtin.include_role:
-        name: cifmw_cephadm
-        tasks_from: check_vip
-      loop: "{{ range(1, (ips | length) + 1) | list }}"
+        - name: Get RGW VIP
+          ansible.builtin.include_role:
+            name: cifmw_cephadm
+            tasks_from: check_vip
+          loop: "{{ range(1, (ips | length) + 1) | list }}"
+          vars:
+            cifmw_cephadm_vip_network: "{{ cifmw_cephadm_rgw_network | default(storage_network_range, true) | default(ssh_network_range, true) }}"
+          when:
+            - cifmw_cephadm_rgw_vip is undefined
+
+        - name: Set RGW VIP
+          ansible.builtin.set_fact:
+            cifmw_cephadm_rgw_vip: "{{ cifmw_cephadm_vip }}/{{ cidr }}"
```

Comment:
I'm ok to keep this VIP with the CIDR and rely on `ansible.utils.ipaddr("address")` to filter the address. However, an alternative approach might be to not have `{{ cidr }}` as part of the VIP, and concatenate that variable (that is set as fact in [1]) directly in the rgw template [2].

[1] https://github.com/openstack-k8s-operators/ci-framework/blob/main/playbooks/ceph.yml#L316
[2] https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/cifmw_cephadm/templates/ceph_rgw.yml.j2#L34

Reply:
I think this might be a cleaner and less error prone approach, because when you think about a VIP you usually refer to an IP address, not a CIDR, while that single variable can be easily concatenated to the rgw template `virtual_ip` parameter.